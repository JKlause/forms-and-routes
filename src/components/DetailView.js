import React from 'react';
import PropTypes from 'prop-types';

const DetailView = ({ photoUrl, name, affiliation, enemies, allies }) => (
  <div>
    {photoUrl && <img src={photoUrl} />}
    {name && <h2>{name}</h2>}
    {affiliation && <span>Affiliation: {affiliation}</span>}
    {enemies && <span>Enemies: {enemies.map(char => `${char} `)}</span>}
    {allies && <span>Allies: {allies.map(char => `${char} `)}</span>}
  </div>

);

DetailView.propTypes = {
  photoUrl: PropTypes.string,
  name: PropTypes.string,
  affiliation: PropTypes.string,
  enemies: PropTypes.array,
  allies: PropTypes.array
};


export default DetailView;