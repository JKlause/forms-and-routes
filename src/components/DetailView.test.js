import React from 'react';
import { shallow } from 'enzyme';
import DetailView from './DetailView';

describe('DetailView component', () => {
  const props = {
    photoUrl: '',
    name: '',
    affiliation: '',
    enemies: [],
    allies: []
  };

  it('renders DetailView', () => {   
    const wrapper = shallow(<DetailView {...props}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
