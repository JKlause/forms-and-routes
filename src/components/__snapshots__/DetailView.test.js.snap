// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailView component renders DetailView 1`] = `
<div>
  <span>
    Enemies: 
  </span>
  <span>
    Allies: 
  </span>
</div>
`;
